docs/src/architecture/notes.md

```diff
@@ -58,8 +63,14 @@ This achieves the following properties:
 - We cannot derive a note's hash from its nullifier.
 - To compute the nullifier, we must know all components of the note: `serial_num`, `script_hash`, `input_hash`, and `vault_hash`.
 
-## Lifecycle of a Note
-For a note to exist, it must be present in the [Notes DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#notes-database) kept by the Miden Node(s). New notes are produced while executing a transaction. They can be produced locally by users in local transactions or by the operator in a network transaction.
+To know a note’s nullifier, one needs to know all details of the note. That means if a note is private and the operator stores only the note's hash, only those with the note details know if this note has been consumed already. Zcash first introduced this approach. 
+
+<p align="center">
+    <img src="../diagrams/architecture/note/Nullifier.png">
```

Comment:
I believe this image is incorrect. The note's hash is not part of the public inputs, that would break the unlikability property (because both the account hash and its nullifier would be public). IMO this distinction is important, and ideally we should fix the diagram

Reply:
I guess it depends on the interpretation. It is true that note hashes are not inputs into transactions, but I don't think this section talks about transactions. The way I interpret this image is that both note hash and note nullifer are publicly known, but the link between them isn't. There is probably a better way to illustrate this though.